ldap/ldap.go

```diff
@@ -8,23 +8,34 @@ import (
 	"net/http"
 
 	"github.com/evergreen-ci/gimlet"
+	"github.com/mongodb/grip"
 	"github.com/pkg/errors"
 	ldap "gopkg.in/ldap.v2"
 )
 
 // userService provides authentication and authorization of users against an LDAP service. It
 // implements the gimlet.Authenticator interface.
 type userService struct {
-	CreationOpts
-	conn ldap.Client
+	url         string
+	port        string
+	userPath    string
+	servicePath string
+	group       string
+	cache       UserCache
+	connect     connectFunc
+	conn        ldap.Client
 }
 
 // CreationOpts are options to pass to the service constructor.
 type CreationOpts struct {
-	URL           string          // URL of the LDAP server
-	Port          string          // Port of the LDAP server
-	Path          string          // Path to users LDAP OU
-	Group         string          // LDAP group to authorize users
+	URL         string // URL of the LDAP server
+	Port        string // Port of the LDAP server
+	UserPath    string // Path to users LDAP OU
+	ServicePath string // Path to service users LDAP OU
+	Group       string // LDAP group to authorize users
+
+	UserCache UserCache
```

Comment:
This breaks the constructor in Evergreen. Are we really going to pass a type that itself implements functions that are also passed to `CreateOpts`?

Reply:
I don't think that's true. you need to either populate this value *or* populate the functions as you currently do in evergreen.

this patch would break evergreen currently because the path variables change, but not for the cache. the "MakeUserCache" method on creation ops does the right thing.